application.js

```diff
@@ -57,31 +54,59 @@ var sessionOptions = {
 
 // find out mongodb connection string from $fh.db
 function run(cb) {
-  sessionInit(sessionOptions, function(err) {
-    if (err) {
-      return cb(err);
-    }
-    // List the user fields which you don't want appearing in the authentication response.
-    // This is being consumed in the raincatcher-user mbaas router.
-    var authResponseExclusionList = ['password'];
-    raincatcherUser.init(mediator, app, authResponseExclusionList, sessionOptions, function(err) {
+  var port = process.env.FH_PORT || process.env.OPENSHIFT_NODEJS_PORT || 8001;
+  var host = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
+
+  // Check if direct mongo connection url exist (available after upgrading db) as mongoStore requires
+  // direct url and will throw an error when trying to connect.
+  if (! process.env.FH_MONGODB_CONN_URL) {
+    idleAndListen(port, host);
```

Comment:
Can this be moved to `lib/sessionInit` and check `sessionOptions` instead of the env var. Else we're spreading mongo-specific initialization around and it'll be harder to reenable redis later.